src/arrayvec.rs

```diff
@@ -1171,15 +1171,20 @@ impl<A: Array> DoubleEndedIterator for ArrayVecIterator<A> {
   #[cfg(feature = "rustc_1_40")]
   #[inline]
   fn nth_back(&mut self, n: usize) -> Option<Self::Item> {
-    let slice =
-      &mut self.data.as_slice_mut()[self.base as usize..self.tail as usize];
-    let n = slice.len().checked_sub(n + 1)?;
-    let item = &mut slice[n];
-
-    /* n is in [0..self.tail - self.base] range, so in u16 range */
-    self.tail = self.base + n as u16;
+    let base = self.base as usize;
+    let tail = self.tail as usize;
+    let slice = &mut self.data.as_slice_mut()[base..tail];
+    let n = n.saturating_add(1);
+
+    if let Some(n) = slice.len().checked_sub(n) {
+      let item = &mut slice[n];
+      /* n is in [0..self.tail - self.base] range, so in u16 range */
```

Comment:
In the future, things like this are a good chance to use `debug_assert`, though we don't need another commit to change that here.